File: src/main/scala/llmTests/a201901a/java/gpt35turbo/GraphImpl.java
//package mmlTests.a201901a.java.gpt35turbo;
//
//import mmlTests.a201901a.java.Graph;
//import mmlTests.a201901a.java.GraphFactory;
//import mmlTests.shared.java.Pair;
//import java.util.ArrayList;
//import java.util.HashSet;
//import java.util.List;
//import java.util.Set;
//import java.util.stream.Stream;
//
//class GraphImpl<X> implements Graph<X> {
//
//    private Set<X> nodes;
//    private Set<Pair<X, X>> edges;
//
//    public GraphImpl() {
//        this.nodes = new HashSet<>();
//        this.edges = new HashSet<>();
//    }
//
//    @Override
//    public Set<X> getNodes() {
//        return nodes;
//    }
//
//    @Override
//    public boolean edgePresent(X start, X end) {
//        return edges.contains(new Pair<>(start, end));
//    }
//
//    @Override
//    public int getEdgesCount() {
//        return edges.size();
//    }
//
//    @Override
//    public Stream<Pair<X, X>> getEdgesStream() {
//        return edges.stream();
//    }
//
//    public void addEdge(X start, X end) {
//        nodes.add(start);
//        nodes.add(end);
//        edges.add(new Pair<>(start, end));
//    }
//}
//

File: src/main/scala/llmTests/a201901a/scala/gpt4/GraphImpl.scala
package llmTests.a201901a.scala.gpt4

import llmTests.a201901a.scala.{Graph, GraphFactory}

import java.util
import java.util.stream.Stream
import llmTests.shared.scala.Pair
import scala.jdk.CollectionConverters._

class GraphImpl[X](val edges: Set[Pair[X, X]]) extends Graph[X] {
  override def getNodes: util.Set[X] =
    edges.flatMap(edge => Seq(edge.fst, edge.snd)).asJava

  override def edgePresent(start: X, end: X): Boolean =
    edges.contains(Pair(start, end))

  override def getEdgesCount: Int =
    edges.size

  override def getEdgesStream: Stream[Pair[X, X]] =
    edges.toSeq.asJava.stream()
}